### PR TITLE
Google Auth

### PR DESCRIPTION
**Pull request by felipesere**
@RabeaGleissner @fabientownsend

### CODE REVIEW

File: lib/advisor/core/login/login_user.ex [hunk 1]
```diff
@@ -0,0 +1,12 @@
+defmodule Advisor.Core.LoginUser do
+  alias Advisor.Core.People
+
+  def find_or_create(%{info: %{email: email, name: name}}) do
+    data = %{name: name, email: email}
```
Comment (fabientownsend) [inline lib/advisor/core/login/login_user.ex]: I think you can inline data here

```
     case People.find_by(email: email) do
       nil -> People.create( %{name: name, email: email})
       user -> user
     end
```

File: lib/advisor/core/people/people.ex [hunk 2]
```diff
@@ -2,6 +2,16 @@ defmodule Advisor.Core.People do
   alias Advisor.Repo
   alias Advisor.Core.Person
   import Ecto.Query
+  import Ecto.Changeset
+
+  def create(data) do
+    %Person{}
+    |> cast(data, [:name, :email])
+    |> validate_format(:email, ~r/@/)
```
Comment (fabientownsend) [inline lib/advisor/core/people/people.ex]: I'm not quite sure it's enough to validate an email! :)

File: lib/advisor_web/authentication/local_strategy.ex [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  def handle_request!(conn) do
+    redirect!(conn, Ueberauth.Strategy.Helpers.callback_path(conn))
+  end
+
+  def info(conn) do
+    %Info{
+      urls: %{website: "8thlight.com"},
+      email: "rswanson@8thlight.com",
+      name: "Ron Swanson"
```
Comment (fabientownsend) [inline lib/advisor_web/authentication/local_strategy.ex]: There is no "Ron Swanson" at 8th Light!

File: lib/advisor_web/controllers/authentication_controller.ex [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+  def callback(%{assigns: %{ueberauth_auth: auth}} = conn, _params) do
+    if auth.info.urls[:website] == "8thlight.com" do
+      target = destination(conn)
+
+      auth
+      |> LoginUser.find_or_create()
+      |> proceed(conn, redirect_to: target)
+    else
+      # Flash?
```
Comment (fabientownsend) [inline lib/advisor_web/controllers/authentication_controller.ex]: Flash sounds like a good idea

File: lib/advisor_web/authentication/local_strategy.ex (hunk 3 above)
Comment (felipesere) [reply lib/advisor_web/authentication/local_strategy.ex]: What! I am sure I've see this person in the office: 
Ron Swanson

**Pull request merged**